packages/replay/src/replay.ts

```diff
@@ -204,6 +204,21 @@ export class ReplayContainer implements ReplayContainerInterface {
         // instead, we'll always keep the last 60 seconds of replay before an error happened
         ...(this.recordingMode === 'error' && { checkoutEveryNms: ERROR_CHECKOUT_TIME }),
         emit: this._handleRecordingEmit,
+        onMutation: (mutations: unknown[]) => {
+          const count = mutations.length;
+
+          if (count > 500) {
+            const breadcrumb = createBreadcrumb({
+              category: 'replay.mutations',
+              message: `A mutation with ${count} changes was recorded, which indicate slow performance.`,
+              data: {
+                mutationsCount: count,
+              },
+            });
+            this._createCustomBreadcrumb(breadcrumb);
+          }
+          return true;
```

Comment:
Just to be sure, will this be shown in the UI automatically or does that need changes in the UI to be reflected?
Do we think the category & message is fine? I basically just freestyled this xD

Reply:
Hmm let me double check the breadcrumbs, iirc we just display them all

Reply:
OK so we have 2 options here: 1) we let UI handle the messaging, or 2) we let SDK handle messaging.

The logic goes like this... if we have `data.label`, it will use that as a title, otherwise it will split the category on `.` and display that as title. Likewise for the body, it will use `message` if `data` does not exist, otherwise it will stringify `data`.

Let's make this a bit more dumb and have the UI determine how it's displayed (since it's easier for us to update the UI).

Reply:
>I'm also thinking if we should only display these crumbs (in the UI) for Sentry internal users only at first - I'm worried this breadcrumb may confuse users?

If we limit it to sentry-internal, let's at least give users a way to opt-in (`_experiments` ?). Maybe someone in https://github.com/getsentry/sentry-javascript/issues/6946 could give it a try and tell us if these crumbs show up in performance-impacted replays. This way we'd have a good indication that we're on the right track to solving this problem.

Reply:
Good point - I'll add an `_experiments` option for this, and will remove message/label for now!